### DISCUSSION TITLE
Is $disconnect() stop all queries?

### DISCUSSION THREAD

**Discussion by littleKitchen** [prisma/prisma#16078] [Q&A]
I am working on a nestjs project. In my case, I need to call $disconnect explicitly to reset the connections with the database because I usually get the error saying that the server has closed the connection (in a particular table). I want to call  $disconnect and do the query again if the initial query is failed.

So, will this interrupt all other queries if I call $disconnect()? (insert, update, ......)

Thank you

**Comment by nurul3101** [comment]
Hey @littleKitchen 👋 

The behaviour of [$disconnect](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/connection-management#disconnect) is to remove all the existing connections with the database. Once you invoke the disconnect call and after that, when a different query is tried to be executed, if there are no existing connections, then a new connection is formed automatically by PrismaClient, so your other queries will not be affected or interrupted.

Do let me know in case you face any other issues.

**Reply by littleKitchen** [reply]
Hi @nurul3101 ,

Assume a query takes 2 sec to get the result. If I call the $disconnect during this period, will this interrupt the query?  

Thank you

**Reply by nurul3101** [reply]
Ideally, $disconnect should be invoked after a query has completed the execution, i.e. after the query response has been sent to the client. If there is an ongoing query and simultaneously at the same time connection is disconnected, I assume the query should fail.

**Reply by pantharshit00** [reply]
👋 Hello @littleKitchen, we’re in the process of archiving our historical GitHub Discussion and I wanted to double check if there is anything else we can do to help in this conversation?
  
Let us know if this question is now answered and, if you can, please share any insights! :raised_hands:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
